ecommerce/api.py

```diff
@@ -1067,12 +1067,21 @@ def get_or_create_data_consent_users(basket):
                 coupon_selection.coupon
             ).payment_version.company
             if company:
+                # We will give priority to the course based consent if one exists, otherwise we'll try to fetch global
+                # consent
                 agreements = (
                     DataConsentAgreement.objects.filter(company=company)
                     .filter(courses__in=courses)
                     .distinct()
                 )
 
+                if not agreements:
+                    # Ideally, There should always be only one global consent agreement for a company at maximum
```

Comment:
Can we log an error if there is more than one global consent agreement found?

Reply:
Added.